python/tvm/relay/backend/contrib/tidl.py

```diff
@@ -0,0 +1,1407 @@
+            self.name_map[self.num_original_outputs + len(self.additional_outputs)] = name
+            self.additional_outputs.append(expr)
+
+    def visit_call(self, call):
+        if isinstance(call.op, Function) and "Compiler" in call.op.attrs and call.op.attrs["Compiler"] == self.compiler:
+            var_map = {}
+            for arg, param in zip(call.args, call.op.params):
+                subgraph_name = "_".join(param.name_hint.split("_")[:2])
+                arg = super().visit(arg)
+                var_map[param] = arg
+                self.add_new_outputs(param.name_hint, arg, was_input=True)
+            new_body = VarReplacer(var_map).visit(call.op.body)
+            # Add subgraph outputs as well
+            self.add_new_outputs(subgraph_name, new_body, was_input=False)
+            return new_body
+        return super().visit_call(call)
+
+    def make_calibration_graph(self, expr):
+        self.num_original_outputs = 1
+        if isinstance(expr.body.checked_type, relay.TupleType):
+            self.num_original_outputs = len(expr.body.checked_type.fields)
+        visit_body = super().visit(expr.body)
+        # Get original output(s)
+        outputs = []
+        if isinstance(visit_body, Tuple):
+            for out in visit_body.fields:
+                outputs.append(out)
+        else:
+            outputs.append(visit_body)
+        # Create new function with added subgraph inputs + outputs
+        return relay.Function(expr.params, relay.Tuple(outputs + self.additional_outputs))
+
+class RemoveMultiplyByOne(ExprMutator):
+    """
+    Removes multiply by 1.0f. This pass when followed by
+    RemoveRedundantTranspose is intended to remove a pattern of
+    Transpose([1, 0]) -> Scale(1.0f) -> Transpose([1, 0]) produced by
+    PyTorch's addmm operator.
+    """
+    def visit_call(self, expr):
+        if expr.op.name == "multiply":
+            if isinstance(expr.args[1], tvm.relay.expr.Constant):
+                data = expr.args[1].data.asnumpy()
+                if data.shape == () and data.item() == 1.0:
+                    return expr.args[0]
+            if isinstance(expr.args[0], tvm.relay.expr.Constant):
+                data = expr.args[0].data.asnumpy()
+                if data.shape == () and data.item() == 1.0:
+                    return expr.args[1]
+        return super().visit_call(expr)
+
+def generate_subgraph_tensors(tidl_target, mod, params, input_node, input_data):
+    """
+    """
+
+    # From partitioned module, create a "calibration model" which can be
+    # executed on CPU and will give additional outputs for boundary tensors.
+    mod_tvm = relay.transform.InferType()(mod)
+    mod_tvm = relay.transform.Inline()(mod_tvm)
+    my_mutator = CalibrationGraphMutator(tidl_target)
+    mod_tvm["main"] = my_mutator.make_calibration_graph(mod_tvm["main"])
+    #print("Calibration module:", mod_tvm)
+    print("Input map:", my_mutator.name_map)

```

Comment:
This print seems unnecessary other than debug mode.

Reply:
Removed.